### DISCUSSION TITLE
Feed URL generator + subscribe

### DISCUSSION THREAD

**Discussion by hogmeade** [FreshRSS/FreshRSS#7103] [Q&A]
**Need**
Tedious task when need to add new feed url frequently. For example, https://nonamewebsite.com/api/SYMBOL.xml where SYMBOL is replaced with needed changing term. What I usually do is to copy the url, then replace the SYMBOL, and add them in subscription management. 

**Solution**
I want to add a line in my Extension's configure.phtml when I only need to enter the SYMBOL into [input text]. It will create the whole feed url following a pattern and add them using "Subscribe Tools API". For example:
nonamewebsite: [input text] [subscribe button]

**Additional context**
I tried to add a field and script handler in the static folder of the extension but seems like the subscribe button I created not triggering any event. Maybe it is not allow. There is secure policy not to add inline scripts directly to the configure.phtml. I tried to use indirectly by putting the script in p/i/scripts and direct the using <script src ...> but not working.
It looks like the JavaScript is running before this button is rendered in the DOM.

**Questions**
Ideally, I would like to have this feed url generator/subscribe button in the extension's configure page. Is there a way to make this work?
If it is forbidden then, where should I put this function without changing the core of FreshRSS?

**Comment by Alkarex** [comment]
I suggest you generate an OPML file with all your feeds, and then import the OPML file in FreshRSS

**Comment by hogmeade** [comment]
Thank you for your prompt response. I just wanted to make it more convenient by having a place that can remember all of the feed url syntax where I can add more feed one by one. For example, stock symbol. It takes more steps to import OPML every time I want to add a symbol to track.
